src/components/EditPartnership/EditPartnership.tsx

```diff
@@ -0,0 +1,115 @@
+import { PartnershipType, UpdatePartnership } from '../../api';
+import { DialogForm, DialogFormProps } from '../Dialog/DialogForm';
+import { DialogState } from '../Dialog/useDialog';
+import {
+  CheckboxesField,
+  CheckboxOption,
+  RadioField,
+  RadioOption,
+  SubmitError,
+} from '../form';
+import { PartnershipCardFragment } from '../PartnershipCard/PartnershipCard.generated';
+import {
+  useDeletePartnershipMutation,
+  useUpdatePartnershipMutation,
+} from './EditPartnership.generated';
+
+interface EditPartnershipProps {
+  state: DialogState;
+  partnership: PartnershipCardFragment;
+}
+
+interface DialogInput {
+  partnership: Except<UpdatePartnership, 'id'>;
+}
+
+export const EditPartnership: FC<EditPartnershipProps> = ({
+  state,
+  partnership,
+}) => {
+  const [updatePartnership] = useUpdatePartnershipMutation();
+  const [deletePartnership] = useDeletePartnershipMutation();
+
+  const submit: DialogFormProps<DialogInput>['onSubmit'] = async (input) => {
+    const inputWithId = {
+      partnership: { ...input.partnership, id: partnership.id },
+    };
+    await updatePartnership({
+      variables: { input: inputWithId },
+      refetchQueries: ['ProjectPartnerships'],
+    });
+  };
+
+  const onDeleteClick = async () => {
+    const { id } = partnership;
+    await deletePartnership({
+      variables: { input: id },
+      refetchQueries: ['ProjectPartnerships'],
+    });
+    state.onClose();
+  };
+
+  const initialValues: DialogInput = {
+    partnership: {
+      agreementStatus: partnership.agreementStatus.value,
+      mouStatus: partnership.mouStatus.value,
+      types: partnership.types.value,
+    },
+  };
+
+  return (
+    <DialogForm
+      onSubmit={submit}
+      {...state}
+      DialogProps={{
+        fullWidth: true,
+        maxWidth: 'xs',
+      }}
+      title={`Edit Partnership with ${partnership.organization.name.value}`}
+      leftAction={<Button onClick={onDeleteClick}>Delete</Button>}
+      initialValues={initialValues}
+    >
+      <SubmitError />
+      <CheckboxesField
+        name="partnership.types"
+        label="Types"
+        row
+        disabled={!partnership.types.canEdit}
+      >
+        {([
+          'Managing',
+          'Funding',
+          'Impact',
+          'Technical',
+          'Resource',
+        ] as PartnershipType[]).map((type: PartnershipType) => (
+          <Box width="50%" key={type}>
+            <CheckboxOption key={type} label={type} value={type} />
+          </Box>
+        ))}
+      </CheckboxesField>
+      {/* TODO: fix missing initial valus and disabled not working on RadioField */}
+      <RadioField
+        name="partnership.agreementStatus"
+        label="Agreement Status"
+        disabled={!partnership.agreementStatus.canEdit}
+      >
+        <RadioOption label="Not Attached" value="NotAttached" />
+        <RadioOption label="Awaiting Signature" value="AwaitingSignature" />
+        <RadioOption label="Signed" value="Signed" />
```

Comment:
These labels should be defined via the `displayPartnershipStatus` function. Could probably loop through like you are doing with partnership type. Also you could move it to a variable so you don't have to do it again below.

Reply:
Used `displayPartnershipStatus` and moved out to variable